consensus/common/src/main/java/org/hyperledger/besu/consensus/common/validator/ValidatorProvider.java

```diff
@@ -28,5 +28,9 @@
 
   Collection<Address> getValidatorsForBlock(BlockHeader header);
 
-  Optional<VoteProvider> getVoteProvider();
+  Optional<VoteProvider> getVoteProviderAtHead();
+
+  default Optional<VoteProvider> getVoteProviderAfterBlock(final BlockHeader header) {
```

Comment:
The alternative was to make QBFT code use the ForkingValidatorProvider directly instead of the interface which also didn't seem great. On top of that, we'd have to refactor/duplicate common code from BftBlockCreatorFactory.createExtraData into QbftBlockCreatorFactory to account for this special case.